src/core/thread/mle.cpp

```diff
@@ -112,6 +111,8 @@ Mle::Mle(Instance &aInstance)
 {
     MeshLocalPrefix meshLocalPrefix;
 
+    mDeviceMode.Set(DeviceMode::kModeRxOnWhenIdle);
```

Comment:
Thanks for identifying this issue!

What about moving the call to `Set()` within the constructor?

https://github.com/openthread/openthread/blob/099333375f4b087ac6063140dc2c4f76db892637/src/core/thread/mle_types.hpp#L348

Reply:
I initially thought about it, but my concern was [here](https://github.com/openthread/openthread/blob/36b16be3a4aebdd9e5d1a18dde984ccff0da8455/src/core/thread/mle.cpp#L3679) - that we won't check exactly what is in the Mode TLV. However, if the bit should be ignored on reception it actually might be better solution. Do you think we should move the `Set` into constructor then?

Reply:
Good points. As you mentioned, the reserved bit should be ignored on reception. We should fix this so that we conform to the spec.